src/App.js

```diff
@@ -53,33 +68,36 @@ const App = () => {
   /**
    * Fetch all country names that have reported cases and set state.
    */
-  async function fetchCountryData() {
+  async function fetchCountryNames() {
     const res = await fetch(`https://covid19.mathdro.id/api/countries`);
     const data = await res.json();
-    const { countries } = await data;
+    const { countries } = data;
 
-    setCountryData(countries.map(country => country.name));
+    setCountryNames(countries.map(country => country.name));
   }
 
   /**
-   * Handle Country change: fetchData and update state.
+   * Handle Country change: update state.
    */
-  async function handleCountryChange(country) {
-    fetchData(country);
+  function handleCountryChange(country) {
     setCountry(country);
   }
 
   useEffect(() => { fetchData() }, []);
   useEffect(() => { fetchDailyData() }, []);
-  useEffect(() => { fetchCountryData() }, []);
+  useEffect(() => { fetchCountryNames() }, []);
+  useEffect(() => { fetchData(country) }, [country]);
```

Comment:
nice, this should run on first render with an `undefined` country too, so perhaps line 86 isn't needed?

Reply:
Yes its not needed 👍